app/Jobs/VerifyBlockJob.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Log;
+use App\Appeal;
+use RuntimeException;
+use App\MwApi\MwApiExtras;
+use Illuminate\Support\Str;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+
+class VerifyBlockJob implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    public $tries = 3;
+
+    private $appeal;
+
+    public function __construct(Appeal $appeal)
+    {
+        $this->appeal = $appeal;
+    }
+
+    public function handle()
+    {
+        if (!MwApiExtras::canEmail($this->appeal->wiki, $this->appeal->getWikiEmailUsername())) {
+            Log::create([
+                'user' => 0,
+                'referenceobject' => $this->appeal->id,
+                'objecttype' => 'appeal',
+                'action' => 'account verification',
+                'reason' => 'user can not be e-mailed thru wiki',
+                'ip' => '127.0.0.1',
+                'ua' => '',
+                'protected' => 0,
+            ]);
+
+            return;
+        }
+
+        $token = Str::random(32);
+
+        $this->appeal->update([
+            'verify_token' => $token,
+        ]);
+
+        $url = url(route('appeal.verifyownership', [$this->appeal, $token]));
```

Comment:
What does this generate on the URL side? It seems like we are trying to send the whole appeal object or am I wrong?

I'm also curious, because if we ensure the token is unique, we won't necessarily need the appeal secret either.

Reply:
`route('appeal.verifyownership', [$this->appeal, $token])` creates the relative url for route that has name `appeal.verifyownership` with parameters `$this->appeal` and `$token`. Laravel knows that we don't want to send the full appeal object and just uses it's id. It results will look something like this: `/appeal/1/verify/abclongtokenhereandsoon`.

`url()` converts that relative URL to absolute url like `http://utrs.test/appeal/1/verify/abclongtokenhereandsoon`